src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/UpToDate/UpToDateCheckStatePersistence.cs

```diff
@@ -223,16 +223,19 @@ private static Dictionary<(string ProjectPath, IImmutableDictionary<string, stri
 
         public int OnAfterCloseSolution(object pUnkReserved)
```

Comment:
I question if close actually needs to actually block on saving the cache.

Reply:
I wondered the same, but was concerned that VS could exit midway through writing the file. Is there some way I can queue the task to hold back shutdown?

Reply:
JoinableTaskFactory that comes from your package block until all pending tasks in its collection have completed on Package.Close (and hence VS shutdown).

But it does not look like CPS's does this for its collection.

Reply:
Generally, our pattern is that you start to store the information during PrioritySolutionClosing event and start saving work in the background after capturing all states requiring UI thread.  So the work can be done in the background during other solution closing work.  When you receive solution close event, waits the background task. In that case, almost all work will be done before solution close, instead of blocking the close work.

Reply:
Addressed in https://github.com/dotnet/project-system/pull/7930